crates/router/src/core/payments/operations/payment_get.rs

```diff
@@ -268,34 +268,35 @@ impl<F: Clone + Send + Sync> Domain<F, PaymentsRetrieveRequest, PaymentStatusDat
         // TODO: do not take the whole payment data here
         payment_data: &mut PaymentStatusData<F>,
     ) -> CustomResult<ConnectorCallType, errors::ApiErrorResponse> {
-        match &payment_data.payment_attempt {
-            Some(payment_attempt) if payment_data.should_sync_with_connector => {
-                let connector = payment_attempt
-                    .connector
-                    .as_ref()
-                    .get_required_value("connector")
-                    .change_context(errors::ApiErrorResponse::InternalServerError)
-                    .attach_printable("Connector is none when constructing response")?;
+        let payment_attempt = &payment_data.payment_attempt;
 
-                let merchant_connector_id = payment_attempt
-                    .merchant_connector_id
-                    .as_ref()
-                    .get_required_value("merchant_connector_id")
-                    .change_context(errors::ApiErrorResponse::InternalServerError)
-                    .attach_printable("Merchant connector id is none when constructing response")?;
+        if payment_data.should_sync_with_connector {
+            let connector = payment_attempt
+                .connector
+                .as_ref()
+                .get_required_value("connector")
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Connector is none when constructing response")?;
 
-                let connector_data = api::ConnectorData::get_connector_by_name(
-                    &state.conf.connectors,
-                    connector,
-                    api::GetToken::Connector,
-                    Some(merchant_connector_id.to_owned()),
-                )
+            let merchant_connector_id = payment_attempt
+                .merchant_connector_id
+                .as_ref()
+                .get_required_value("merchant_connector_id")
                 .change_context(errors::ApiErrorResponse::InternalServerError)
-                .attach_printable("Invalid connector name received")?;
+                .attach_printable("Merchant connector id is none when constructing response")?;
 
-                Ok(ConnectorCallType::PreDetermined(connector_data.into()))
-            }
-            None | Some(_) => Ok(ConnectorCallType::Skip),
+            let connector_data = api::ConnectorData::get_connector_by_name(
+                &state.conf.connectors,
+                connector,
+                api::GetToken::Connector,
+                Some(merchant_connector_id.to_owned()),
+            )
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Invalid connector name received")?;
+
+            Ok(ConnectorCallType::PreDetermined(connector_data.into()))
```

Comment:
```suggestion
            Ok(ConnectorCallType::PreDetermined(SomeType::from(connector_data)))
```
Makes code navigation easier.